PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/webapp/WebAppBasePropertyView.java

```diff
@@ -0,0 +1,431 @@
+import com.microsoft.tooling.msservices.serviceexplorer.azure.webapp.WebAppBasePropertyMvpView;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.webapp.WebAppPropertyViewPresenter;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.webapp.base.WebAppBasePropertyViewPresenter;
+
+public abstract class WebAppBasePropertyView extends BaseEditor implements WebAppBasePropertyMvpView {
+    public final String id;
+    protected final WebAppBasePropertyViewPresenter presenter;
+    private final Map<String, String> cachedAppSettings;
+    private final Map<String, String> editedAppSettings;
+    private final StatusBar statusBar;
+
+    private static final String PNL_OVERVIEW = "Overview";
+    private static final String PNL_APP_SETTING = "App Settings";
+    private static final String BUTTON_EDIT = "Edit";
+    private static final String BUTTON_REMOVE = "Remove";
+    private static final String BUTTON_ADD = "Add";
+    private static final String TABLE_HEADER_VALUE = "Value";
+    private static final String TABLE_HEADER_KEY = "Key";
+    private static final String TXT_NA = "N/A";
+    private static final String TABLE_LOADING_MESSAGE = "Loading ... ";
+    private static final String TABLE_EMPTY_MESSAGE = "No available settings.";
+    private static final String FILE_SELECTOR_TITLE = "Choose Where You Want to Save the Publish Profile.";
+    private static final String NOTIFY_PROPERTY_UPDATE_SUCCESS = "Properties updated.";
+    private static final String NOTIFY_PROFILE_GET_SUCCESS = "Publish Profile saved.";
+    private static final String NOTIFY_PROFILE_GET_FAIL = "Failed to get Publish Profile.";
+    private static final String INSIGHT_NAME = "AzurePlugin.IntelliJ.Editor.WebAppBasePropertyView";
+
+    private JPanel pnlMain;
+    private JButton btnGetPublishFile;
+    private JButton btnSave;
+    private JButton btnDiscard;
+    private JPanel pnlOverviewHolder;
+    private JPanel pnlOverview;
+    private JPanel pnlAppSettingsHolder;
+    private JPanel pnlAppSettings;
+    private JTextField txtResourceGroup;
+    private JTextField txtStatus;
+    private JTextField txtLocation;
+    private JTextField txtSubscription;
+    private JTextField txtAppServicePlan;
+    private HyperlinkLabel lnkUrl;
+    private JTextField txtPricingTier;
+    private JTextField txtJavaVersion;
+    private JTextField txtContainer;
+    private JTextField txtContainerVersion;
+    private JLabel lblJavaVersion;
+    private JLabel lblContainer;
+    private JLabel lblContainerVersion;
+    private JBTable tblAppSetting;
+    private DefaultTableModel tableModel;
+    private AnActionButton btnAdd;
+    private AnActionButton btnRemove;
+    private AnActionButton btnEdit;
+
+    protected WebAppBasePropertyView(@NotNull Project project, @NotNull String sid,
+                                     @NotNull String resId, @Nullable String slotName) {
+        this.id = getId();
+        this.presenter = createPresenter();
```

Comment:
just as I mentioned, also attaching the view here. 
`createPresenter()` only helps you to create and specify the instance of the presenter.

Reply:
Yes, attach view in base class works.